README.md

```diff
@@ -1291,6 +1292,186 @@ let release () =
+
+Here is an example function that allows itself to be interrupted:
+
+```ocaml
+let run_op ~interrupted =
+  Fiber.first
+    (fun () ->
+       Eio.Condition.await_no_mutex interrupted;
+       traceln "Cancelled at user's request."
+    )
+    (fun () ->
+       traceln "Running operation (Ctrl-C to cancel)...";
+       Fiber.await_cancel ()       (* Simulated work *)
+    )
+```
+
+Note that we don't need a mutex here.
+We're just waiting for the number of interrupts received to change,
+and, since that increases monotonically, once we get woken we always want to continue.
+Also, we don't care about missing interrupts from before this operation started.
+
+The code here is quite subtle.
+We rely on the fact that the first branch of the `Fiber.first` runs first,
+and only starts running the second branch once `await_no_mutex` has finished registering.
+Thus, we never display the message telling the user to press Ctrl-C before we're ready
+to receive it.
+This isn't likely to matter if a human is responding to the message,
+but if the response is automated then the delay could matter.
+
+To run this function, we need to install a signal handler.
+There are very few things that you can do safely in a signal handler.
+For example, you can't take a mutex in a signal handler
+because the signal might have interrupted a fiber that had already locked it.
+However, you can safely call `Eio.Condition.broadcast`:
+
+<!-- $MDX non-deterministic=command -->
+```ocaml
+# Eio_main.run @@ fun _env ->
+  let interrupted = Eio.Condition.create () in
+  let handle_signal (_signum : int) =
+    (* Warning: we're in a signal handler now.
+       Most operations are unsafe here, except for Eio.Condition.broadcast! *)
+    Eio.Condition.broadcast interrupted
+  in
+  Sys.set_signal Sys.sigint (Signal_handle handle_signal);
+  run_op ~interrupted;;
++Running operation (Ctrl-C to cancel)...
+[ user presses Ctrl-C here ]
++Cancelled at user's request.
+- : unit = ()
+```
+
+Another common pattern when using signals is using `SIGHUP`
+to tell an application to reload its configuration file:
+
+<!-- $MDX file=examples/signals/main.ml,part=main -->
+```ocaml
+let main ~config_changed =
+  while true do
+    Fiber.both
+      (fun () ->
+         (* First, we start waiting for SIGHUP.
+            This is so that if we get SIGHUP before we finish loading
+            the old configuration then we'll start again. *)
+         Eio.Condition.await_no_mutex config_changed;
+         traceln "Received SIGHUP";
+         (* We could cancel the loading fiber now, in case it's still running,
+            but in this example we just wait for it to finish by itself. *)
+      )
+      (fun () ->
+         traceln "Reading configuration ('kill -SIGHUP %d' to reload)..." (Unix.getpid ());
+         Eio_unix.sleep 2.0;    (* Simulate loading config *)
+         traceln "Finished reading configuration";
+      )
+  done
+```
```

Comment:
I was wonder if it might be worth pointing the user how he can then communicate outside that the configuratoin changed and was reloaded, this assumes the configuration is stored in some global reference, but that's probably too much, so it's fine as it is.

Reply:
I added some comments to the full example, but just factored this out as `load_config ()` in the README, as it's not very relevant to handling signals.